pgsqltoolsservice/connection/connection_service.py

```diff
@@ -119,19 +119,12 @@ def connect(self, params: ConnectRequestParams) -> Optional[ConnectionCompletePa
             if cancellation_key in self._cancellation_map:
                 self._cancellation_map[cancellation_key].cancel()
             self._cancellation_map[cancellation_key] = cancellation_token
-
-        # Map the connection options to their psycopg2-specific options
-        connection_options = {CONNECTION_OPTION_KEY_MAP.get(option, option): value for option, value in params.connection.options.items()
-                              if option in PG_CONNECTION_PARAM_KEYWORDS}
-
-        # Use the default database if one was not provided
-        if 'dbname' not in connection_options or not connection_options['dbname']:
-            connection_options['dbname'] = self._service_provider[constants.WORKSPACE_SERVICE_NAME].configuration.pgsql.default_database
-
-        # Connect using psycopg2
+        
+        # Get the type of server
+        provider_name = self._get_provider_name(params.owner_uri)
         try:
-            # Pass connection parameters as keyword arguments to psycopg2.connect by unpacking the connection_options dict
-            connection = psycopg2.connect(**connection_options)
+            # Get connection to DB server using the provided connection params
+            connection: ServerConnection = DriverManager(provider_name, params).get_connection()
```

Comment:
Could this be ConnectionManager instead o DriverManager ?

Reply:
Done!